doc/api/n-api.md

```diff
@@ -2298,6 +2298,27 @@ Returns `napi_ok` if the API succeeded.
 This API attempts to delete the `key` own property from `object`.
 
 
+#### *napi_has_own_property*
+<!-- YAML
+added: REPLACEME
+-->
+```C
+napi_status napi_has_own_property(napi_env env,
+                                  napi_value object,
+                                  napi_value key,
+                                  bool* result);
+```
+
+- `[in] env`: The environment that the N-API call is invoked under.
+- `[in] object`: The object to query.
+- `[in] key`: The name of the own property whose existence to check.
+- `[out] result`: Whether the own property exists on the object or not.
+
+Returns `napi_ok` if the API succeeded.
+
+This API checks if the Object passed in has the named own property.
```

Comment:
Can we add a note about side effects? E.g., what if the object is a revoked proxy? If the key is not to name?